env/env_exporter.py

```diff
@@ -14,85 +14,88 @@
-            elif opt == 'jreplica':
-                return 'openebs/'+'jiva'+':'+jivaDetails[1]
-    file.close() 
+
+
+def get_jiva_detail(commit, option, *args, **kwargs):
+    '''
+        It takes commit as an input and return corresponding docker_image_name with tag
+        If the option doesn't belongs to Jiva it will return None
+        args and kwargs can be used for passing additional variable and futher customizing the function
+    '''
+    jivaDetails = ['jiva', commit.rstrip('\n')]
+    if option == 'jcontroller':
+        return 'openebs/jiva:'+jivaDetails[1]
+    if option == 'jreplica':
+        return 'openebs/jiva:'+jivaDetails[1]
     return None
 
-# cstor
-def getCstor(filename):
-    file = open(filename, "r")
-    for line in file:
-        data = line.split(',')
-        # print(data)
-        repo = data[2].split('=') #data[2] has repo info
-        commit = data[4].split('=') # data[4] has commit info
-        if repo[1] == 'cstor':
-            cstorDetails = ['cstor', commit[1].rstrip('\n')]
-            file.close()
-            return cstorDetails
-    file.close() 
+
+def get_cstor_detail(commit, option, *args, **kwargs):
+    '''
+    It takes commit as an input and return corresponding docker_image_name with tag
+    If the option doesn't belongs to Cstor it will return None
+    args and kwargs can be used for passing additional variable and futher customizing the function
+    '''
+    return ['cstor', commit.rstrip('\n')] if option == 'cstor' else None
+
+
+def get_docker_image_name_with_tag(filename, option):
+    content = None
+    storage_eng_detail_map = {
+        'maya': get_maya_detail,
+        'jiva': get_jiva_detail,
+        'cstor': get_cstor_detail
+    }
+
+    with open(filename, 'r') as f:
+        content = f.readlines()
+
+    # Iterate over each commit message
+    for line in content:
+        # sample line: date=10-12-18,time=18:45:13,repo=maya,branch=master,commit=20364c17
+        _, _, repo, branch, commit = line.split(',')
+        _, repo_name = repo.split('=')
+        _, commit_hash = commit.split('=')
+        # getting the corresponding function that can be used to get docker_img_name_with_tag for the repo
+        get_docker_img_name_func = storage_eng_detail_map[repo_name]
+        data = get_docker_img_name_func(commit_hash, option)
+        if data:
+            # if the commit belongs to the options that is provided in input.
+            return data
     return None
 
+
 def run(option):
-    filename = "../baseline/baseline"
-    if (option == 'mapi') or (option == 'iovolume'):
-        maya = getMaya(filename, option)
-        print maya
-        return
-    elif (option == 'jcontroller') or (option == 'jreplica'):
-        jiva = getJiva(filename, option)
-        print jiva
-        return
-    elif option == 'cstor':
-        cstor = getCstor(filename)
-        print cstor
-        return
+    filename = '../baseline/baseline'
```

Comment:
Can you create a flag that takes file path as an argument?

Reply:
Sure.

Reply:
Hey I updated the PR. Please have a look.